### ISSUE TITLE
Op.eq in where clauses is broken in TypeScript 4.4

### ISSUE THREAD

**Issue by PurkkaKoodari** [sequelize/sequelize#13505]
## Issue Creation Checklist

- [x] I have read the [contribution guidelines](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)

## Bug Description

`Op.eq` and `Op.is` (and possibly others) cause a compile error when placed in `where` clauses in TypeScript 4.4 and up. TypeScript 4.1 through 4.3 work fine.

I assume this is due to the fact that `WhereOperators` doesn't include them as valid keys. Why it worked earlier, I have no idea.

https://github.com/sequelize/sequelize/blob/main/types/lib/model.d.ts#L141

### SSCCE

https://github.com/PurkkaKoodari/sequelize-sscce

(note that I don't expect this code to run properly due to a missing Sequelize instance, just to compile)

```typescript
import {DataTypes, Model, Op, Sequelize} from 'sequelize';

const sequelize: Sequelize = null as any;

interface PostAttributes {
  authorId: number;
}

class Post extends Model<PostAttributes> implements PostAttributes {
  public authorId!: number;
};

Post.init({
  authorId: DataTypes.INTEGER,
}, {
  sequelize,
});

Post.findAll({
  where: {
    authorId: {
      [Op.eq]: 2
    }
  }
});
```

### What do you expect to happen?

The code should compile as it's literally copy-pasted from https://sequelize.org/master/manual/model-querying-basics.html#the-basics

### What is actually happening?

```
error TS2322: Type '{ [Op.eq]: number; }' is not assignable to type 'string | number | bigint | boolean | Fn | Col | Literal | Where | WhereAttributeHash<any> | Date | Buffer | WhereOperators | ... 6 more ... | undefined'.
  Types of property '[Op.eq]' are incompatible.
    Type 'number' is not assignable to type 'undefined'.
```

on the `findAll` call.

### Environment

- Sequelize version: 6.6.5 
- Node.js version: 14.17.6 
- TypeScript version: at least 4.4.0-beta through 4.4.3 

### How does this problem relate to dialects?

- [x] I think this problem happens regardless of the dialect.
- [ ] I think this problem happens only for the following dialect(s): 
- [ ] I don't know, I was using PUT-YOUR-DIALECT-HERE, with connector library version XXX and database version XXX

### Would you be willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time and I know how to start.
- [x] Yes, I have the time but I don't know how to start, I would need guidance.
    - I can definitely go and add the operators to `WhereOperators` if they should be there.
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

**Comment by l2ysho** [comment]
#13553 waiting for approval

**Comment by l2ysho** [comment]
@remcohaszing I am still waiting to PR approve, I dont have mental strenght to update PR with `Op.col`, I suggest to create another PR for `Op.col`

**Comment by ksze** [comment]
This seems to still be broken in 6.9.0. What's going on?

**Comment by WikiRik** [comment]
https://github.com/sequelize/sequelize/pull/13553 was merged after 6.9.0 was released. It will become available with the next release

**Comment by information-security** [comment]
@ksze 
This is released in [`6.10.0`](https://www.npmjs.com/package/sequelize/v/6.10.0). You can upgrade the package now.
